engine/src/cmd/script/script_variables.cpp

```diff
@@ -457,7 +457,7 @@ void Mission::doSetVar(missionNode *node, int mode) {
             }
             vi = global_var->script.varinst;
         }
-        if (vi->type != VAR_BOOL || vi->type != VAR_FLOAT || vi->type != VAR_INT || vi->type != VAR_OBJECT) {
+        if (vi->type != VAR_BOOL && vi->type != VAR_FLOAT && vi->type != VAR_INT && vi->type != VAR_OBJECT) {
```

Comment:
this might be slightly more performant if a switch was used instead
```
swtich(vi->type) {
case VAR_BOOL:  // fall-thru
case VAR_FLOAT:  // fall-thru
case VAR_INT:  // fall-thru
case VAR_OBJECT:
    // do nothing
    break;
default:
    fatalError(node, node, "unsupported type in setvar");
    assert(0); // is there a better method of existing?
}

Reply:
Good idea

Reply:
Done. Implemented it that way.